### PR TITLE
Added Hidden type filter

### PR DESCRIPTION
**Pull request by orendig**

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/hide_types.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  ENABLE_HIDE_TYPES_IN_NACLS,
+} from '../types'
+import {
+  isCustomObject,
+} from '../transformers/transformer'
+
+const filterCreator: FilterCreator = ({ config }) => ({
+  onFetch: async (elements: Element[]): Promise<void> => {
+    elements
+      .filter(isType)
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/hide_types.ts:32]: I think you need `isObjectType` here

Comment (orendig) [reply packages/salesforce-adapter/src/filters/hide_types.ts:32]: but also primitive and basically also list...

Comment (adamro) [reply packages/salesforce-adapter/src/filters/hide_types.ts:32]: Lists are never top level elements so not really relevant. The primitive types that are added in SF are a question, do we want to hide them? (makes sense if we do, I just didn't think about it)

Comment (orendig) [reply packages/salesforce-adapter/src/filters/hide_types.ts:32]: https://github.com/salto-io/salto/blob/a5963689af9ff60098f220b7873d22e8b1138e66/packages/workspace/src/workspace/hidden_values.ts#L30
isType is the definition of hidden type... Yes we want (and need) to hide the primitives in SF and we want to supporting list as well.

Comment (adamro) [reply packages/salesforce-adapter/src/filters/hide_types.ts:32]: 👍

**Pull request merged**